src/mac_features/nrf_802154_ifs.c

```diff
@@ -101,60 +107,66 @@ static bool is_ifs_needed_by_address(const uint8_t * p_frame)
     return false;
 }
 
-/**@brief Checks if the IFS is needed by measuring time between the actual and the last frames. */
-static bool is_ifs_needed_by_time(const uint8_t * p_frame, bool * is_lifs)
+/**@brief Checks if the IFS is needed by measuring time between the actual and the last frames. 
+ *        Returns the needed ifs, 0 if none.
+ */
+static uint16_t ifs_needed_by_time(const uint8_t * p_frame, uint32_t current_timestamp)
 {
-    assert(m_current_timestamp > m_last_frame_timestamp);
-    uint32_t dt = m_current_timestamp - m_last_frame_timestamp;
-
+    assert(nrf_802154_timer_sched_time_is_in_future(m_last_frame_timestamp, 0, current_timestamp));
```

Comment:
I understand you idea here, but this assert can be detrimental.
Consider the case, when last frame has been transmitted so long ago (more than 2^31 us), that `nrf_802154_timer_sched_time_is_in_future` returns false due to integer overflow. Then the code asserts here, but it should't.

Proposed solutions:
- Simple. Just remove the assert. It is better to add one unwanted ifs delay if device transmits very rarely and go on than hang due to assert.
- Design change. Start IFS timer when previous frame has been transmitted. If the timer fires and no new transmit request has been requested during its period, the timer does nothing. If there is a new transmit request when timer is counting, check matching adresses. If no match, kill timer and transmit, If match, just save some static data to make timer's callback transmit when it fires.

@jciupis @hubertmis Please comment your opinion

Reply:
I'm in favour of dropping the assert completely - as @ankuns said, it's not a big deal to send a frame with several hundred microseconds delay if the previous one was sent tens of minutes before that (and that's how long the timer scheduler takes to overflow, give or take).

Reply:
I agree that the problem exists. Its probability of occurence is quite low (maybe only Green Power device would send the frames rarely), but nevertheless it is a problem due to the bad design.

Now, about the resolutions.
Removing assert is trivial, naturally.
Your proposed design change would do the job, however it would make implementation less straightforward for the sake of complexity. Also I would suspect an increased power consumption (every transmission means one more callback to kick, so more CPU clicking and so on).

Let's wait for the comments.

Reply:
I agree we cannot assert it here due to reason pointed by @ankuns .
How about following code?
```suggestion
    if (!nrf_802154_timer_sched_time_is_in_future(m_last_frame_timestamp, 0, current_timestamp))
    {
        return 0;
    }
```

Reply:
Hubert's resolution is fine by me. Everybody agrees?

Reply:
*(in Aragorn's voice)* You have my sword!

Reply:
ok, go with code proposed by @hubertmis